torch_geometric/datasets/web_qsp_dataset.py

```diff
@@ -0,0 +1,251 @@
+# Alot of code in this file is based on the original G-Retriever paper
+# url: https://arxiv.org/abs/2402.07630
+from typing import Dict, List, Tuple, no_type_check
+
+import numpy as np
+import torch
+from tqdm import tqdm
+
+from torch_geometric.data import Data, InMemoryDataset
+from torch_geometric.nn.nlp import SentenceTransformer
+
+try:
+    from pandas import DataFrame
+    WITH_PANDAS = True
+except ImportError:
+    DataFrame = None
+    WITH_PANDAS = False
+
+try:
+    from pcst_fast import pcst_fast
+    WITH_PCST = True
+except ImportError:
+    WITH_PCST = False
+
+try:
+    import datasets
+    WITH_DATASETS = True
+except ImportError:
+    WITH_DATASETS = False
+
+
+@no_type_check
+def retrieval_via_pcst(
+    graph: Data,
+    q_emb: torch.Tensor,
+    textual_nodes: DataFrame,
+    textual_edges: DataFrame,
+    topk: int = 3,
+    topk_e: int = 3,
+    cost_e: float = 0.5,
+) -> Tuple[Data, str]:
+    c = 0.01
+    if len(textual_nodes) == 0 or len(textual_edges) == 0:
+        desc = textual_nodes.to_csv(index=False) + "\n" + textual_edges.to_csv(
+            index=False, columns=["src", "edge_attr", "dst"])
+        graph = Data(x=graph.x, edge_index=graph.edge_index,
+                     edge_attr=graph.edge_attr, num_nodes=graph.num_nodes)
+        return graph, desc
+
+    root = -1  # unrooted
+    num_clusters = 1
+    pruning = "gw"
+    verbosity_level = 0
```

Comment:
```suggestion
```

Reply:
its not clear to me why we need to remove this?  these are used later in the function
```
    vertices, edges = pcst_fast(edges, prizes, costs, root, num_clusters,
                                pruning, verbosity_level)
```

Reply:
nit: it just looked unnecessary to me to assign literals to local variables, do nothing on the variables and feed them to `pcst_fast` :D

Reply:
oh i see what you mean, yes i can move these into the function definition for cleanliness, i misunderstood the request. will do

Reply:
okay i actually looked into this. the pcst-fast function is defined as so:
```
TypeError: pcst_fast(): incompatible function arguments. The following argument types are supported:
    1. (arg0: numpy.ndarray[numpy.int64], arg1: numpy.ndarray[numpy.float64], arg2: numpy.ndarray[numpy.float64], arg3: int, arg4: int, arg5: str, arg6: int) -> tuple[numpy.ndarray[numpy.int32], numpy.ndarray[numpy.int32]]

Invoked with: array([[   0,    1],
       [   2,    0],
       [   3,    4],
       ...,
       [1722, 1567],
       [ 551, 1723],
       [1723,   15]]), array([0.        , 0.        , 0.        , ..., 0.16666669, 0.15000004,
       0.33333337], dtype=float32), array([0.5, 0.5, 0.5, ..., 0. , 0. , 0. ]); kwargs: root=-1, num_clusters=1, pruning='gw', verbosity_level=0
```

therefore something like
```
    vertices, edges = pcst_fast(edges, prizes, costs, root=-1, num_clusters=1,
                                pruning="gw", verbosity_level=0)
```
fails because theres no kwarg support. and so the cleanest way to pass these values while making it clear what they mean is the original way. will leave as is